### PR TITLE
Remove special casing from AnalysisConfig

### PR DESCRIPTION
**Pull request by mortalisk**
While working on a parsing prototype, found that there is some code paths that in only used in testing. Removing those and updating tests.

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/analysis_config.py [hunk 1]
```diff
@@ -49,11 +49,7 @@ def __init__(
 
     def _copy_modules(self):
         for element in self._analysis_copy:
-            if isinstance(element, list):
-                src_name, dst_name = element
-            else:
-                src_name = element[ConfigKeys.SRC_NAME]
-                dst_name = element[ConfigKeys.DST_NAME]
+            src_name, dst_name = element
```
Comment (andreas-el) [inline src/ert/_c_wrappers/enkf/analysis_config.py:52]: I take it that we should always get lists of tuples from now on.
But should we then express this in the __init__ using optionals ?

``` python
def __init__(
        self,
        ...
        analysis_copy=None,
        analysis_set_var=None,
        analysis_select=None,
    ):
```

**Review by andreas-el — CHANGES_REQUESTED** [review]

File: src/ert/_c_wrappers/enkf/analysis_config.py (hunk 1 above)
Comment (mortalisk) [reply src/ert/_c_wrappers/enkf/analysis_config.py:52]: what() ?

Comment (andreas-el) [reply src/ert/_c_wrappers/enkf/analysis_config.py:52]: Discussed and was decided not to follow up on.

**Review by andreas-el — APPROVED** [review]
Shoot for the stars. :rocket:

**Pull request merged**